www_src/components/basic-element/basic-element.jsx

```diff
@@ -78,31 +85,46 @@ var El = React.createClass({
     if (this.props.type === 'link') {
       this.positionButton();
     }
+
+    // Since basic elements never unmount, we need to make sure that
+    // zIndex updates, which may occur outside of regular element
+    // interaction, are reflected back into the element's state.
+    //
+    // TODO: this discovery might mean we need to do this for all
+    //       transform properties too, but there haven't been any
+    //       bugs filed about those (yet...?)
+    if(this.props.zIndex !== this.state.zIndex) {
+      this.setState({
+        zIndex: this.props.zIndex
+      });
+    }
+  },
```

Comment:
This is the actual fix. Everything else is refactors and cleanup to get to this point, and while it's tempting to say "let's refactor separately", I **strongly** urge us not to. Let's take this refactor as necessary for this fix to even exist.